llpc/context/llpcCompiler.cpp

```diff
@@ -2271,6 +2277,46 @@ void helperThreadBuildRayTracingPipelineElf(IHelperThreadProvider *helperThreadP
   helperThreadPayload->compiler->releaseContext(context);
 }
 
+// =====================================================================================================================
+// Limited implementation of Llpc::IHelperThreadProvider to support -add-rt-helpers.
+//
+// If no deferred work helper thread providers is available when additional threads are requested via -add-rt-helpers
+// then use an instances of this class to coordinate helper threads.
+class InternalHelperThreadProvider : public Llpc::IHelperThreadProvider {
+public:
+  InternalHelperThreadProvider() {}
+  ~InternalHelperThreadProvider() {}
+
+  virtual void SetTasks(ThreadFunction *pFunction, uint32_t numTasks, void *pPayload) override {
+    m_totalInstances.exchange(numTasks);
```

Comment:
This seems to only work if `SetTasks` is only called once on a `InternalHelperThreadProvider` object. Maybe assert that `m_totalInstances` is zero to enforce that? (Or instead handle multiple calls by initializing the other fields as well.)

Reply:
Yes, this class is meant to be single use, so we don't need m_totalInstances atomic at all.  Added assert.